blockchain/api.go

```diff
@@ -753,9 +753,29 @@ func (api *StandardTokenApi) AllowanceOf(ctx context.Context, from, to common.Ad
 	return api.tokenContract.Allowance(getCallOptions(ctx), from, to)
 }
 
-func (api *StandardTokenApi) Approve(ctx context.Context, key *ecdsa.PrivateKey, to common.Address, amount *big.Int) (*types.Transaction, error) {
+func (api *StandardTokenApi) Approve(ctx context.Context, key *ecdsa.PrivateKey, to common.Address, amount *big.Int) error {
 	opts := api.opts.getTxOpts(ctx, key, api.opts.gasLimit)
-	return api.tokenContract.Approve(opts, to, amount)
+	// reset allowance
+	tx, err := api.tokenContract.Approve(opts, to, big.NewInt(0))
```

Comment:
nope =(

Reply:
Why? Setting allowance without resetting it to zero value doesn't work now, why not to perform this in `Approve()` method?

Reply:
this not obvious behavior for method definition.
if you want, you may returning error, that allowance couldn't changed by this way.

Reply:
This makes API less obvious to use, and you ALWAYS need to reset allowance, so you will be forced to write a bit more **similar** code before **any** call of the `.Approve()` method. 

I'd prefer to describe this behavior in the method's docstrings and add extra handling for a case when you need to set allowance to zero.